### ISSUE TITLE
Feature Request: Share Button 

### ISSUE THREAD

**Issue by desputinski** [friendica/friendica#5785]
Hi, I would like to use a friendica-share button on some of my websites.

Now it is possible to share Links with the diaspora-share button for friendica with this service you can reach here:
https://share.diasporafoundation.org/?title=diaspora*%20Advanced%20Sharer

Maybe one of you developers has time to clone that code:
https://share.diasporafoundation.org/about.html

Now, if I use the (functioning) diaspora-share-button people get confused if they click on a friendica share button but land on the diaspora sharing site....

**Comment by annando** [comment]
I would like to have some "Fediverse" share button that would work with many different systems. This I would prefer over a Friendica-only solution.

**Comment by tobiasd** [comment]
You could try https://gitlab.com/Makhi/sharer for your page. - see [here](http://darkpractices.net/sharer/en/index.html?text=The%20Internet%20is%20our%20Social%20Network&url=https://friendi.ca) for a running installation with the EN translation.

**Comment by desputinski** [comment]
Hi tobiasd,
thanks for the link! but I don't understand how I can implement this on my website. Which files do I link how to a share button? I have a wordpress website....

**Comment by MrPetovan** [comment]
Simply have a share icon with the following hypertext link:
```html
<a href="http://darkpractices.net/sharer/{Language}/?text={Description}&url={URL}" target="sharer">
  <img src="..." alt="Share"/>
</a>
```
Where:
- `{Language}` is a two-letter language code (`en`, `fr`, `de`, etc...).
- `{Description}` is a short description of the page that will be added in the shared post. It could simply be the title in the case of an article.
- `{URL}` is the URL to share, most probably the current page URL.

**Comment by desputinski** [comment]
Hi MrPetovan,
thanks! But for me this didn't work.
See example here:
https://neoliberalyse.de/TTIP/index.html

**Comment by desputinski** [comment]
Hm that also doesn't work. now I have
http://darkpractices.net/sharer/en/?text={Description}&url={URL}

I am forwarded to the sharing site but nothing is filled out, so no url is coming with it and no description. So this is button is no use...

**Comment by desputinski** [comment]
See https://neoliberalyse.de/TTIP/index.html

**Comment by tobiasd** [comment]
Do you have addons active in your browser, preventing the execution of javascript from 3rd party like cdn.jquery.com?

**Comment by desputinski** [comment]
I tried three different browsers. Does it work in your browser if you click on my html file I linked?

**Comment by MrPetovan** [comment]
It does, and it shows `{URL}` pre-filled out in the sharing window. As for `{Language}`, you have to replace it with the actual URL you want to share. Same with `{Description}`.

**Comment by desputinski** [comment]
I don't understand that. Strange: look:
https://gesundheitstabelle.de/images/browser.mp4

**Comment by MrPetovan** [comment]
In your code, replace `{URL}` by `https://neoliberalyse.de/TTIP/index.html` and `{Description}` by `test` (the title of the page). The `Host` part is for the sharer to fill since it's the Friendica host you want to share on.

**Comment by desputinski** [comment]
Ok, so it doesn't work automatically to fill in the url and the description, but it should? Because if the sharer has to fill it in manually the share button is more or less useless, I suppose :)

**Comment by MrPetovan** [comment]
There are two different actors here:
- The Publisher who exposes the Share button on any page they are editing. Could be a Wordpress blog, a self-hosted website, anything.
- The Sharer who reads the page and could want to share it on their Friendica node.

It is up to the publisher to format the sharing link with the appropriate URL and description in advance. Currently it has to be manual on any page the sharing button is on, but we can imagine a Wordpress plugin to automate the replacement of the page URL and title into the sharing link.

If the sharing link has been properly edited with the URL and Description, then the Sharer only has to fill in their Friendica node hostname, as the two other field will have been pre-populated thanks to the Publisher work.

**Comment by desputinski** [comment]
Hm, this sharing-button passes on the url and the title without me having to do anything more than to put the code on my site:
https://share.diasporafoundation.org/about.html

**Comment by tobiasd** [comment]
Assuming you have your images in `/img` on your server and in that directory an image called `sharer.png` this HTML snipplet should work to have a button filling the needed fields automatically via javascript (code is essentially taken from the linked page for the d* share button. I just replaced the URL that is called to match the test installation of Makhi. If you are running a high traffic page, you might want to host the sharer yourself.).

```html
<a href="javascript:;" onclick="window.open('http://darkpractices.net/sharer/en/index.html?url='+encodeURIComponent(location.href)+'&text='+encodeURIComponent(document.title),'das','location=no,links=no,scrollbars=no,toolbar=no,width=620,height=550'); return false;" rel="nofollow" target="_blank" style="padding: 3px 4px 2px; background: #fafafa; border: 1px solid #ddd; font-size: 13px; color: #222; text-decoration: none;">
  <img src="/img/sharer.png" style="width: 32px; height: 32px; padding: 2px;"> share this
</a>
```

It fills the URL and the page title for the sharer. You have to enter the URL of the instance you are using. If you have a different image location in mind, you have to change the `src="...` part of the 2nd last line. If you have no image at all, remove the `<img ...>` part completely.

**Comment by desputinski** [comment]
Hi tobiasd, thank you!
That already works great. I changed the code a little bit to this 
[code]
<a href="javascript:;" onclick="window.open('http://darkpractices.net/sharer/en/index.html?url='+encodeURIComponent(location.href)+'&text='+encodeURIComponent(document.title),'das','location=no,links=no,scrollbars=yes,toolbar=no,width=620,height=650'); return false;" rel="nofollow" target="_blank" style="padding: 3px 4px 2px; background: #fff; border: 1px solid #ddd; font-size: 13px; color: #222; text-decoration: none;"><img src="https://politikvonunten.org/wp-content/uploads/2018/09/Teilen-Dienste.png"></a>[/code]

To be perfect I would like to kick all the services out I don't need. Like mastadon. Do you know, if that is easy to do?

**Comment by tobiasd** [comment]
That can be easily done by hosting the sharer code yourself. It is a simple HTML page where you could remove the services you don't want to offer. But why not allow the visitors to share your link on other services?

**Comment by desputinski** [comment]
Hi tobiasd,
thank you! Now I translated it in German and changed the order of the social networks. The complete file can be downloaded here: https://politikvonunten.org/teilen/teilen.zip

**Comment by desputinski** [comment]
Hi, just one more question.
I wanted to modify this to have a preset Friendica-Pod/Server already in the form. Now I have this as code:
`<input type="text" class="form-control host" id="hostFriendica" value="https://sozial.politikvonunten.org">`
The problem is, that with this the form is already filled out, but the blue "share" button is not active. So I have to click into the Server/Pod-Form-Field and modify it a little and after that I'm able to click "share".
Any ideas, why the share button get's active only if I have manually edited the form field?
See: https://politikvonunten.org/teilen-politikvonunten/

**Comment by MrPetovan** [comment]
Yes, see the JS code: https://gitlab.com/Makhi/sharer/blob/master/assets/js/sharer.js

The form input gets disabled only on `keyup`, when something is typed.

**Comment by desputinski** [comment]
So does that mean I cannot get rid of this function so the form-input is taken no matter if it is filled out or not? I could also maybe code the url into the js-file? It's on my server now anyway...

**Comment by MrPetovan** [comment]
Yes, I encourage you to modify the original script to accept a new URL parameter `host` that would be pre-filled in in form. You can copy what was done for `text` and `url`.

**Comment by desputinski** [comment]
I can't really do javascript. Could you tell me, what exactly I have to change in the sharer.js? That would be great!

**Comment by MrPetovan** [comment]
Sure, the first part should look like this:
```javascript
	//get url parameter
	var varText = GetURLParameter('text');
	var varUrl  = GetURLParameter('url');
	var varId   = GetURLParameter('articleid');
	var varHost = GetURLParameter('host'); // <-- Added
	$('.text').val(decodeURIComponent(varText));
	$('.url').val(decodeURIComponent(varUrl));
	$('.host').val(decodeURIComponent(varHost)); // <-- Added
```

**Comment by desputinski** [comment]
Thanks :) 
But it didn't work. If I add those two lines I get "undefined" in the top form field.

**Comment by MrPetovan** [comment]
Ha, yes, it's more complicated than this. I won't take the time to rewrite the whole script. I have a better idea. If you want to allow sharing only on a specific Friendica node, here's the relevant snippet:

```HTML
<a href="javascript:;" onclick="window.open('http://<your.friendica.host>/bookmarklet?url='+encodeURIComponent(location.href)+'&title='+encodeURIComponent(document.title), 'das', 'location=no,links=no,scrollbars=yes,toolbar=no,width=620,height=650'); return false;" rel="nofollow" target="_blank" style="padding: 3px 4px 2px; background: #fff; border: 1px solid #ddd; font-size: 13px; color: #222; text-decoration: none;">
    <img src="https://politikvonunten.org/wp-content/uploads/2018/09/Teilen-Dienste.png">
</a>
```
This bypasses the sharer completely and directly submits the link to the hardcoded Friendica host.

**Comment by desputinski** [comment]
Thank you that seems to work!

**Comment by MrPetovan** [comment]
The only limitation is that it doesn't allow to share on a different Friendica node.

**Comment by desputinski** [comment]
doesn't matter. I will have two share buttons. One for all alternative networks, and one just for this one pod/server

**Comment by desputinski** [comment]
HI, I found a bug in the share button: If I click I get a friendica-Window and am able to log on to friendica. But the Link is not shared directyl if I'm not already logged in.
So I have to click on the share button once again to share it. This behaviour is ok for me but it would confuse and scare off any Facebookuser nor non techie...

**Comment by MrPetovan** [comment]
I can reproduce this, after log in the user isn't redirected back to the `/bookmarklet` module but the `/network` module.

**Comment by MrPetovan** [comment]
I've never looked into that part of the code, so I'm not exactly sure how easy/fast it would be, but it should be doable.

**Comment by desputinski** [comment]
Hm, so I have to contact the developers of darkpractices, I suppose...

**Comment by MrPetovan** [comment]
There's nothing they can do on their end, we have to improve our redirection after login system.

**Comment by Extarys** [comment]
Since #5818 is closed, what is left to do here? If someone have time to explain what needs to be done maybe I can help.

**Comment by MrPetovan** [comment]
Someone™ needs to test that the bookmarklet works as intended now.

**Comment by ikkqqstudio** [comment]
https://share.joinmastodon.org/
